server/src/main/java/io/crate/execution/dml/FloatVectorIndexer.java

```diff
@@ -72,20 +70,11 @@ public FloatVectorIndexer(Reference ref) {
     }
 
     @Override
-    public void indexValue(float @Nullable [] values,
-                           XContentBuilder xcontentBuilder,
+    public void indexValue(float @NotNull [] values,
                            Consumer<? super IndexableField> addField,
+                           TranslogWriter translogWriter,
                            Synthetics synthetics,
                            Map<ColumnIdent, ColumnConstraint> toValidate) throws IOException {
-        if (values == null) {
-            return;
-        }
-        xcontentBuilder.startArray();
```

Comment:
Passing a float array directly to XContentBuilder.value() has the same effect as writing the entries out individually.